dashboard/modules/job/job_manager.py

```diff
@@ -556,21 +573,66 @@ async def _monitor_job_internal(
         self, job_id: str, job_supervisor: Optional[ActorHandle] = None
     ):
         is_alive = True
-        if job_supervisor is None:
-            job_supervisor = self._get_actor_for_job(job_id)
-
-            if job_supervisor is None:
-                logger.error(f"Failed to get job supervisor for job {job_id}.")
-                await self._job_info_client.put_status(
-                    job_id,
-                    JobStatus.FAILED,
-                    message="Unexpected error occurred: Failed to get job supervisor.",
-                )
-                is_alive = False
 
         while is_alive:
             try:
-                await job_supervisor.ping.remote()
+                job_status = await self._job_info_client.get_status(job_id)
+                if job_status == JobStatus.PENDING:
+                    # Compare the current time with the job start time.
+                    # If the job is still pending, we will set the status
+                    # to FAILED.
+                    job_info = await self._job_info_client.get_info(job_id)
+                    timeout = float(
+                        os.environ.get(
+                            RAY_JOB_START_TIMEOUT_SECONDS_ENV_VAR,
+                            DEFAULT_JOB_START_TIMEOUT_SECONDS,
+                        )
+                    )
+                    if job_info.start_time in [None, 0]:
+                        logger.error(
+                            f"Job {job_id} start time not found. "
+                            f"Skipping job start timeout check."
+                        )
+                    if time.time() - job_info.start_time / 1000 > timeout:
+                        err_msg = (
+                            "Job supervisor actor failed to start within "
```

Comment:
Can you also write the action items here? Most likely this happens because the job submission requires the resources? Can you write action items regarding this? Is there any other case this can happen?

Reply:
I don't think there's any other case where it can happen.  Added a message